tests/testthat/test-learn.R

```diff
@@ -1,74 +1,86 @@
+  order_of_deletion <- c("tf_2", "tf_1", "tf_5", "tf_6", "tf_4", "tf_3")
+  remote_tbl_names <- map_chr(
+    order_of_deletion,
+    ~dbplyr::remote_name(dm_for_filter_copied[[.x]])
+  ) %>% set_names(order_of_deletion)
+
+  withr::defer(
+    walk(
+      dm_get_tables_impl(dm_for_filter_copied)[order_of_deletion],
+      ~ dbExecute(src_db$con, paste0("DROP TABLE ", dbplyr::remote_name(.x)))
+    )
+  )
+
+  dm_for_filter_db_learned_all <- dm_from_src(src_db)
+
+  # in case there happen to be other tables in schema "dbo" or "public"
+  dm_for_filter_db_learned <-
+    dm_for_filter_db_learned_all %>%
+    dm_select_tbl(!!!remote_tbl_names)
+
+  expect_equivalent_dm(
+    dm_for_filter_db_learned,
+    dm_for_filter()[order_of_deletion]
   )
 })
 
-# dm_learn_from_postgres() --------------------------------------------------
-test_that("Learning from Postgres works?", {
-  src_postgres <- skip_if_error(src_test("postgres"))
-  con_postgres <- src_postgres$con
 
-  # create an object on the Postgres-DB that can be learned
-  if (is_postgres_empty()) {
-    copy_dm_to(con_postgres, dm_for_filter(), temporary = FALSE, table_names = unique_db_table_name)
-  }
+test_that("Learning from specific schema on MSSQL or Postgres works?", {
 
-  dm_for_filter_postgres_learned <- dm_from_src(src_postgres)
-  dm_for_filter_postges_learned_from_con <- dm_from_src(con_postgres)
+  skip_if_src_not(c("mssql", "postgres"))
+  src_db <- my_test_src()
+  con_db <- src_db$con
 
-  dm_postgres_learned_renamed <-
-    dm_rename_tbl(
-      dm_for_filter_postgres_learned,
-      !!!set_names(src_tbls(dm_for_filter_postgres_learned), src_tbls(dm_for_filter()))
-    )
+  schema_name_q <- DBI::dbQuoteIdentifier(con_db, schema_name)
 
-  dm_postgres_learned_from_con_renamed <-
-    dm_rename_tbl(
-      dm_for_filter_postges_learned_from_con,
-      !!!set_names(src_tbls(dm_for_filter_postges_learned_from_con), src_tbls(dm_for_filter()))
-    )
+  DBI::dbExecute(con_db, paste0("CREATE SCHEMA ", schema_name_q))
 
-  expect_equivalent_dm(
-    dm_postgres_learned_renamed,
-    dm_for_filter()
+  dm_for_disambiguate_copied <- copy_dm_to(
+    src_db,
+    dm_for_disambiguate(),
+    temporary = FALSE,
+    table_names = ~ DBI::SQL(dbplyr::in_schema(schema_name_q, .x))
   )
+  order_of_deletion <- c("iris_3", "iris_2", "iris_1")
+  remote_tbl_names <- set_names(paste0(schema_name_q, order_of_deletion), order_of_deletion)
+
+  withr::defer(
+    {
+      walk(
+        remote_tbl_names,
+        ~ dbExecute(con_db, paste0("DROP TABLE ", .x))
+      )
+      dbExecute(con_db, paste0("DROP SCHEMA ", schema_name_q))
+    }
+  )
+
+  dm_for_filter_db_learned <- dm_from_src(src_db, schema = schema_name) %>%
+    dm_select_tbl(!!!order_of_deletion)
 
   expect_equivalent_dm(
-    dm_postgres_learned_from_con_renamed,
-    dm_for_filter()
+    dm_for_filter_db_learned,
+    dm_for_disambiguate()[order_of_deletion]
```

Comment:
Why is `dm_for_filter_...` compared with `dm_for_disambiguate()` ? Also:

```suggestion
    dm_for_disambiguate()
```

Reply:
good catch :) I must have just copied & pasted (in the end it was a `dm_for_disambiguate` in disguise).
Don't know about reordering, see above.